src/cluster_facil/utils_auto_label.py

```diff
@@ -0,0 +1,145 @@
+"""
+Funções utilitárias para rotulação automática de clusters usando modelos de linguagem (ex: GPT-4.1-nano via OpenAI API).
+"""
+import os
+from typing import List, Optional, Dict
+from openai import OpenAI
+import json
+
+def gerar_rotulo_cluster(
+    cluster_texts: List[str],
+    model: str = "gpt-4.1-nano",
+    api_key: Optional[str] = None,
+    temperature: float = 0.0,
+) -> str:
+    """
+    Gera um rótulo (nome/tema) para um cluster de textos usando a OpenAI Responses API.
+    Args:
+        cluster_texts (List[str]): Lista de textos pertencentes ao cluster.
+        model (str): Nome do modelo OpenAI a ser utilizado. Padrão: 'gpt-4.1-nano'.
+        api_key (Optional[str]): Chave da API OpenAI. Se não fornecida, busca em OPENAI_API_KEY.
+        temperature (float): Temperatura do modelo (criatividade).
+    Returns:
+        str: Rótulo sugerido para o cluster.
+    """
+    if not api_key:
+        api_key = os.getenv("OPENAI_API_KEY")
+    if not api_key:
+        raise ValueError("É necessário fornecer uma chave de API OpenAI via argumento ou variável de ambiente OPENAI_API_KEY.")
+
+    client_kwargs = {"api_key": api_key}
+    client = OpenAI(**client_kwargs)
+
+    sample_texts = cluster_texts[:10]
+    # Delimita cada texto com <sampleN>...</sampleN>
+    joined_texts = "\n\n".join(f"<sample{i+1}>\n{t}\n</sample{i+1}>" for i, t in enumerate(sample_texts))
+    prompt = (
+        "Dado o seguinte conjunto de textos, gere um rótulo curto (tema) que represente o cluster. "
+        "O rótulo deve ser claro, conciso e descritivo."
+    )
```

Comment:
```suggestion
    prompt = (
        "Dado o seguinte conjunto de textos, gere um rótulo curto (tema) que represente o cluster. "
        "O rótulo deve ser claro, conciso e descritivo.\n"
        "Se não for possível identificar um padrão claro entre as decisões, responda "falta_coesão".
    )
```

Uma parte importante das rodadas de clusterização é entender quais clusters estão coesos e quais não estão. Acho que precisamos indicar isso explicitamente como uma possibilidade, para que esses casos possam ser novamente clusterizados.

Nesse sentido, seria interessante eventualmente incluir a possibilidade de que uma segunda rodada de auto_label seja automaticamente aplicada apenas nos casos considerados não coesos. E, a médio prazo, poderiamos automatizar seguidas rodadas de reclusterização.

Ligado a isso, registro aqui algum ceticismo em relação a se modelos menores vão ser capazes de fazer essa decisão de falta de unidade. Acho que os modelos menores vão se sentir mais pressionados a dizer que há algo em comum, pelo que já tive de experiência em outros casos.